api/src/main/java/org/pmiops/workbench/genomics/GenomicExtractionMapper.java

```diff
@@ -26,13 +31,24 @@
   @Mapping(target = "submissionDate", source = "dbSubmission.terraSubmissionDate")
   GenomicExtractionJob toApi(DbWgsExtractCromwellSubmission dbSubmission);
 
-  default TerraJobStatus convertJobStatus(FirecloudSubmissionStatus status) {
-    if (status == FirecloudSubmissionStatus.DONE) {
+  default TerraJobStatus convertJobStatus(FirecloudSubmission submission) {
+    // FirecloudSubmission.status doesn't capture the distinction between 'succeeded' and 'failed'
```

Comment:
Would we be able to drive our TerraJobStatus off of just the workflow status and not look at the submission status at all?

Following the logic like in here, https://github.com/DataBiosphere/terra-ui/blob/fa71713c7633485839121538960fd81cc6f376ed/src/components/job-common.js#L10. Although in our case, "submitted" would just be "Running".

Reply:
Yeah, probably could, since we know for this specific wdl that there will only be one workflow.